### PR TITLE
Instances for date and time types

### PR DESCRIPTION
**Pull request by edsko**

### CODE REVIEW

File: src/Cardano/Prelude/GHC/Heap/NormalForm/Classy.hs [hunk 1]
```diff
@@ -390,6 +391,17 @@ deriving via UseIsNormalForm Word16 instance NoUnexpectedThunks Word16
...
 
+deriving via UseIsNormalForm Day              instance NoUnexpectedThunks Day
+deriving via UseIsNormalForm DiffTime         instance NoUnexpectedThunks DiffTime
+deriving via UseIsNormalForm LocalTime        instance NoUnexpectedThunks LocalTime
+deriving via UseIsNormalForm NominalDiffTime  instance NoUnexpectedThunks NominalDiffTime
+deriving via UseIsNormalForm TimeLocale       instance NoUnexpectedThunks TimeLocale
+deriving via UseIsNormalForm TimeOfDay        instance NoUnexpectedThunks TimeOfDay
+deriving via UseIsNormalForm TimeZone         instance NoUnexpectedThunks TimeZone
+deriving via UseIsNormalForm UniversalTime    instance NoUnexpectedThunks UniversalTime
+deriving via UseIsNormalForm UTCTime          instance NoUnexpectedThunks UTCTime
```
Comment (dnadales) [inline src/Cardano/Prelude/GHC/Heap/NormalForm/Classy.hs]: GitHub told me to try multi-line comments :man_shrugging:

File: src/Cardano/Prelude/GHC/Heap/NormalForm/Classy.hs [hunk 2]
```diff
@@ -390,6 +391,17 @@ deriving via UseIsNormalForm Word16 instance NoUnexpectedThunks Word16
 deriving via UseIsNormalForm Word32 instance NoUnexpectedThunks Word32
 deriving via UseIsNormalForm Word64 instance NoUnexpectedThunks Word64
 
+deriving via UseIsNormalForm Day              instance NoUnexpectedThunks Day
+deriving via UseIsNormalForm DiffTime         instance NoUnexpectedThunks DiffTime
```
Comment (mrBliss) [inline src/Cardano/Prelude/GHC/Heap/NormalForm/Classy.hs]: I need this :+1:

File: src/Cardano/Prelude/GHC/Heap/NormalForm/Classy.hs (hunk 1 above)
Comment (edsko) [reply src/Cardano/Prelude/GHC/Heap/NormalForm/Classy.hs]: :roll_eyes:

**Comment by mrBliss** [comment]
Superseded by #81

**Pull request closed without merging**